src/_data/cookbook/design_patterns.yml

```diff
@@ -0,0 +1,74 @@
+- title: "Persistent storage architecture: Key-value data"
+  tags:
+    - data
+    - shared-preferences
+    - dark mode
+  description: Save application data to a user's on-device key-value store.
+  path: /cookbook/architecture/key-value-data
+  img: /assets/images/docs/app-architecture/design-patterns/kv-store-icon.svg
+  body: |
+    Most Flutter applications, no matter how small or big they are, 
+    require storing data on the user’s device at some point, 
+    such as API keys, user preferences or data that should be available offline.
+    
+    In this recipe, you will learn how to integrate persistent storage for key-value
+    data in a Flutter application by implementing a switchable light-mode and dark-mode theme 
+    in an application and storing the users preference.
```

Comment:
In some SSGs, there are ways to extract the data of a page during the build process and use that data in templates. I assume its possible in 11ty, but I need to learn about how 11ty works before I can look into that. WDYT?

Reply:
Can you explain a bit more what data you'd want to access and where you want to use it? Definitely sounds possible :)

Reply:
Rather than having to write a yml file with the data, you could access the frontmatter and content of any given page during the build process by its path. So in the the `design-patterns.md` file you could write something like:

```
assign pages = pageNodes.where(path.contains('/cookbook/architecture')
for page in pages:
  var intro = page.content.takeChars(500)
  render expansionPanel(title: paage.frontmatter.title, body: intro, etc...)
```

This assumption that its _possible_ is based on my experience (long ago) with Gatsby. Gatsby is, for better or worse, extremely feature rich and not light-weight at all, so as I type this I'm realizing that this may be an unrealistic expectation any given SSG.

In Gatsby you can configure multiple build steps, so the first step can take all the data from your markdown files and put it into an object that's available globally in your templates. Given that the data from these yml files are available in the templates, I'm wondering if its possible to access the md content as well if we configure 11ty in some way.

Reply:
I've watched some 11ty videos this morning, and I think this is possible using the tags/collections feature. I don't want to implement it in this PR, but that feature might allow us to clean up the website code and content in the future. I'm still learning about 11ty and learning the implementation details of our website, so take this all with several grains of salt.

Reply:
Nevermind, its really straight-forward, I implemented it in this PR.